core/arch/interp.c

```diff
@@ -4224,6 +4224,8 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
 #endif
     }
     else if (dcontext->forged_exception_addr == bb->start_pc) {
+        /* Might have been cleared by client_process_bb. */
```

Comment:
Is this the right way to do ?
i.e. calling `bb_process_single_step` during the loop over instructions, and then trying to keep the flags that we set